mlx/coverity_services.py

```diff
@@ -1,614 +1,380 @@
+            matcher_type (str): The type of the matcher (nameMatcher, idMatcher or keyMatcher)
 
-        # create page spec
-        page_spec = self.client.factory.create('pageSpecDataObj')
-        page_spec.pageSize = 9999
-        page_spec.sortAscending = True
-        page_spec.startIndex = 0
+        Returns:
+            dict: New filter for API queries
+        """
+        matchers = []
+        for filter_ in filter_values:
+            matcher = {"type": matcher_type}
+            if matcher_type == "nameMatcher":
+                matcher["name"] = filter_
+                matcher["class"] = "Component"
+                assert column_name == "component"
+            elif matcher_type == "idMatcher":
+                matcher["id"] = filter_
+            else:
+                matcher["key"] = filter_
+            matchers.append(matcher)
 
-        # create snapshot scope
-        snapshot_scope = self.client.factory.create('snapshotScopeSpecDataObj')
+        if column_name not in self.columns:
+            self.logger.warning(f"Invalid column name {column_name!r}; Retrieve column keys first.")
 
-        snapshot_scope.showOutdatedStreams = False
-        snapshot_scope.compareOutdatedStreams = False
+        return {
+            "columnKey": self.columns[column_name],
+            "matchMode": "oneOrMoreMatch",
+            "matchers": matchers
+        }
 
-        snapshot_scope.showSelector = 'last()'
-        snapshot_scope.compareSelector = 'last()'
+    def get_defects(self, stream, filters, column_names):
+        """Gets a list of defects for given stream, filters and column names.
+        If a column name does not match the name of the `columns` property, the column can not be obtained because
+        it need the correct corresponding column key.
+        Column key `cid` is always obtained to use later in other functions.
 
-        logging.info('Running Coverity query...')
-        return self.client.service.getMergedDefectsForSnapshotScope(project_id, filter_spec,
-                                                                    page_spec, snapshot_scope)
+        Args:
+            stream (str): Name of the stream to query
+            filters (dict): Dictionary with attribute names as keys and CSV lists of attribute values to query as values
+            column_names (list[str]): The column names
 
-    def handle_attribute_filter(self, attribute_values, name, *args, **kwargs):
-        """ Applies any filter on an attribute's values.
+        Returns:
+            dict: The content of the request. This has a structure like:
+                {
+                    "offset": 0,
+                    "totalRows": 2720,
+                    "columns": [list of column keys]
+                    "rows": list of [list of dictionaries {"key": <key>, "value": <value>}]
+                }
+        """
+        logging.info("Querying Coverity for defects in stream [%s] ...", stream)
```

Comment:
Should we use `self.logger` instead?

Reply:
If you want this info logged, yes

Reply:
Uhm, no, not really. But if logging.info has no effect, we should remove it or replace it with self.logger.debug for example.

Reply:
And make it possible to enable debug logs with a configuration variable in conf.py.

Reply:
Is it maybe possible to set it via sphinx make command as well?

Reply:
Resolved in 5e964aafb19adc8f1bf78ac69e1ebfa7d0878654

Reply:
`LOGLEVEL` can be used to set the logging level to `INFO` for example.
57b267b2e0235cc0de3bc6891ee65822af7638af